### PR TITLE
[fix](planner) fix bug in agg on constant column

### PR DESCRIPTION
**Pull request by englefly**
s# Proposed changes
For performance reason, we want to remove constant column from groupingExprs.
For example:
                `select sum(T.A) from T group by T.B, 'xyz'` is equivalent to `select sum(T.A) from T group by T.B`
We can remove constant column `abc` from groupingExprs.

But there is an exception when all groupingExpr are constant
For example:

                sql1: `select 'abc' from t group by 'abc'`
                 is not equivalent to
                sql2: `select 'abc' from t`

                sql3: `select 'abc', sum(a) from t group by 'abc'`
                 is not equivalent to
                sql4: `select 1, sum(a) from t`
                (when t is empty, sql3 returns 0 tuple, sql4 return 1 tuple)

We need to keep some constant columns if all groupingExpr are constant.

Consider sql5 `select a from (select "abc" as a, 'def' as b) T group by b, a;`
if the constant column `a` is in select list, this column should not be removed.
sql5 is transformed to 
sql6 `select a from (select "abc" as a, 'def' as b) T group by a;`

Issue Number: close #xxx

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 35.25 seconds
 load time: 480 seconds
 storage size: 17170512191 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230206083620_clickbench_pr_90790.html

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java [hunk 1]
```diff
@@ -1291,11 +1291,54 @@ private void analyzeAggregation(Analyzer analyzer) throws AnalysisException {
...
+                        groupingExprs.clear();
+                        groupingExprs.addAll(tempExprs);
+                        //consider sql1, groupingExprs need at least one expr, it can be
+                        // any original grouping expr. we use the first one.
+                        if (groupingExprs.isEmpty()) {
+                            groupingExprs.add(theFirstConstantGroupingExpr);
+                        }
+                    }
+                } else {
+                    groupingExprs.removeIf(Expr::isConstant);
```
Comment (carlvinhust2012) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java:1341]: here， if ‘aggExprs.isEmpty()’ not true，no need add an expr to 'groupingExprs'?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java [hunk 2]
```diff
@@ -1291,11 +1291,54 @@ private void analyzeAggregation(Analyzer analyzer) throws AnalysisException {
...
+                        if (keep) {
+                            tempExprs.add(groupExpr);
+                        } else {
+                            someGroupExprRemoved = true;
+                        }
+                    }
+                    if (someGroupExprRemoved) {
+                        groupingExprs.clear();
+                        groupingExprs.addAll(tempExprs);
+                        //consider sql1, groupingExprs need at least one expr, it can be
```
[the remark this answers is not in the document]
Comment (englefly) [reply fe/fe-core/src/main/java/org/apache/doris/analysis/SelectStmt.java]: Thanks!

**Comment by englefly** [comment]
@starocean999 PTAL. This pr is related to ur previous fix.

File: regression-test/suites/trino_p0/constant_group_key.groovy [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    explain {
+        sql("select 'oneline' from nation group by 'constant1', 'constant2'")
+        contains "group by: 'constant1'"
+    }
+
+    //no group by key
+    explain {
+        sql("select 'oneline', count(*) from nation group by 'constant1', 'constant2'")
+        contains "group by: \n"
```
Comment (morrySnow) [inline regression-test/suites/trino_p0/constant_group_key.groovy:33]: scalar agg is not same with vector agg, so we could not remove all constant group by exprs.

File: regression-test/suites/trino_p0/sql/group_by/runConstantGroupBy_order.sql [hunk 4]
```diff
@@ -1 +1,4 @@
-select 2 from nation group by 1
+select 2 from nation group by 1;
+select "oneline" from nation group by "I am a const, but not to be removed";
```
Comment (morrySnow) [inline regression-test/suites/trino_p0/sql/group_by/runConstantGroupBy_order.sql:2]: we need to add a regression test on empty table to set scalar agg and vector agg's results.

**Review by morrySnow — DISMISSED** [review]

File: regression-test/suites/trino_p0/sql/group_by/runConstantGroupBy_order.sql (hunk 4 above)
Comment (englefly) [reply regression-test/suites/trino_p0/sql/group_by/runConstantGroupBy_order.sql:2]: done

**Review by morrySnow — APPROVED** [review]

**Pull request merged**